Userland/Utilities/strings.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+#include <unistd.h>
+
+enum class StringOffsetFormat {
+    None = 0,
+    Decimal,
+    Octal,
+    Hexadecimal
+};
+
+static bool should_print_characters(Vector<u8> const& characters)
+{
+    for (u8 ch : characters) {
+        if (is_ascii_printable(ch) && !is_ascii_space(ch))
+            return true;
+    }
+    return false;
+}
+
+static void print_characters(Vector<u8> const& characters, StringOffsetFormat string_offset_format, size_t string_offset_position)
+{
+    switch (string_offset_format) {
+    case StringOffsetFormat::Decimal:
+        outln("offset {:d}: ", string_offset_position);
```

Comment:
Is the newline at the end (since you are using `outln` instead of `out`) intentional? Same applies for the other two appearances of `outln` that are tasked to print the offset.

Also, if we want to go for compatibility with the POSIX standard, we should drop the `offset ` prefix as well as the `0o` and `0x` prefixes.

Reply:
> Is the newline at the end (since you are using `outln` instead of `out`) intentional?

It was intentional. However I can understand if we want things to be more POSIX-y, the latest commit matches the expected output format.

Reply:
Was there a point in switching away from `out` there? If you were going to create a string builder to format and store a single string with the offset, you might as well `.append` the data there as well.

Reply:
IMO it was less ugly to have a single `outln` at the end of the function instead of 3 separate `out`s in the switch statement. However, in the interest of moving things along, I have changed it to the latter.